File: 2848. Points That Intersect With Cars/Solution.cpp
#include <algorithm>
#include <ranges>
#include <vector>
using namespace std;

class Solution {
public:
    int numberOfPoints(vector<vector<int>>& nums)
    {
        int max_end = ranges::max(nums, {}, [](const vector<int>& x) { return x[1]; })[1];
        vector<int> diff(max_end + 2, 0);
        for (const vector<int>& p : nums) {
            ++diff[p[0]];
            --diff[p[1] + 1];
        }
        int res = 0, s = 0;
        for (int d : diff) {
            s += d;
            res += s > 0;
        }
        return res;
    }
};

File: 2848. Points That Intersect With Cars/Solution.py
from itertools import accumulate
from typing import List


class Solution:
    def numberOfPoints(self, nums: List[List[int]]) -> int:
        max_end = max(end for _, end in nums)
        diff = [0] * (max_end + 2)
        for start, end in nums:
            diff[start] += 1
            diff[end + 1] -= 1
        return sum(x > 0 for x in accumulate(diff))
